msdk compile failed due to incorrect reference to cl.hpp

•	https://github.com/KhronosGroup/OpenCL-CLHPP/tree/v2.0.11/include/CL - CL/ch.hpp is there
•	https://github.com/KhronosGroup/OpenCL-CLHPP/tree/v2.0.12/include/CL - it is missed

And this was done within https://github.com/KhronosGroup/OpenCL-CLHPP/pull/100.

opencl-cpp-headers now comes with cl2.hpp https://github.khronos.org/OpenCL-CLHPP/

rpm -qf /usr/include/CL/cl2.hpp
opencl-cpp-headers-2.0.12-7.1.noarch

the error in question
intel-mediasdk-20.3~pre/samples/sample_plugins/rotate_opencl/include/sample_opencl_plugin.h:50:10: fatal error: CL/cl.hpp: No such file or directory
#include "CL/cl.hpp"

--- a/samples/sample_plugins/rotate_opencl/include/opencl_filter.h
+++ b/samples/sample_plugins/rotate_opencl/include/opencl_filter.h
@@ -1,5 +1,5 @@
 /******************************************************************************\
-Copyright (c) 2005-2019, Intel Corporation
+Copyright (c) 2005-2020, Intel Corporation
 All rights reserved.
 
 Redistribution and use in source and binary forms, with or without modification, are permitted provided that the following conditions are met:
@@ -21,7 +21,18 @@ or https://software.intel.com/en-us/media-client-solutions-support.
 
 #include <iostream>
 #include <stdexcept>
-#include <CL/cl.h>
+#if defined(_WIN32) || defined(_WIN64)
+  // There is no cl2.hpp in Intel OpenCL SDK for Windows at least up to 2020.1.396
+  #include <CL/cl.h>
+#else
+  #define CL_HPP_ENABLE_SIZE_T_COMPATIBILITY
+  #define CL_HPP_ENABLE_PROGRAM_CONSTRUCTION_FROM_ARRAY_COMPATIBILITY
+  #define CL_HPP_ENABLE_EXCEPTIONS
+  #define CL_HPP_CL_1_2_DEFAULT_BUILD
+  #define CL_HPP_MINIMUM_OPENCL_VERSION 120
+  #define CL_HPP_TARGET_OPENCL_VERSION 120
+  #include <CL/cl2.hpp>
+#endif
 
 #include "mfxvideo++.h"
 #include "logger.h"

--- a/samples/sample_plugins/rotate_opencl/src/sample_opencl_plugin.cpp
+++ b/samples/sample_plugins/rotate_opencl/src/sample_opencl_plugin.cpp
@@ -1,5 +1,5 @@
 /******************************************************************************\
-Copyright (c) 2005-2019, Intel Corporation
+Copyright (c) 2005-2020, Intel Corporation
 All rights reserved.
 
 Redistribution and use in source and binary forms, with or without modification, are permitted provided that the following conditions are met:
@@ -646,10 +646,10 @@ void OpenCLRotator180Context::CreateBuffers(const cl::size_t<3> &Y_size,
 
 void OpenCLRotator180Context::SetKernelArgs()
 {
-    m_kernelY .setArg(0, m_InY());
-    m_kernelY .setArg(1, m_OutY());
-    m_kernelUV.setArg(0, m_InUV());
-    m_kernelUV.setArg(1, m_OutUV());
+    m_kernelY .setArg(0, m_InY);
+    m_kernelY .setArg(1, m_OutY);
+    m_kernelUV.setArg(0, m_InUV);
+    m_kernelUV.setArg(1, m_OutUV);
 }
 
 void OpenCLRotator180Context::Rotate(size_t width, size_t height,

